dataplicity/m2m/bencode.py

```diff
@@ -60,12 +61,19 @@ def encode(obj):
 
     def add_encode(obj):
         if isinstance(obj, bytes):
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                append(six.b("{}:".format(len(obj))))
+                append(obj)
         elif isinstance(obj, text_type):
             obj = obj.encode('utf-8')
-            append(b"{}:{}".format(len(obj), obj))
+            if PY2:
+                append(b"{}:{}".format(len(obj), obj))
+            elif PY3:
+                add_encode(obj)
```

Comment:
The PY2 detection should probably be moved out of the inner loop and in to the module scope. Or, better still, written without branching on the PY version. This version has sacrificed a bit of performance for compatibility, which we should avoid here. It matters because all our packets go through this code.